### PR TITLE
Add export feature

### PR DESCRIPTION
**Pull request by 0livd**
This adds the possibility to export the bills and the transactions of a project as json or csv
The exported fields are the following:
transactions: amount, receiver, ower
bills: what, payer_name, payer_weight, owers, amount, date

The link to export the project is located in the options menu

Original feature request #28

### CODE REVIEW

File: budget/models.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def balance(self):
     def uses_weights(self):
         return len([i for i in self.members if i.weight != 1]) > 0
 
-    def get_transactions_to_settle_bill(self):
+    def get_transactions_to_settle_bill(self, pretty_output=False):
         """Return a list of transactions that could be made to settle the bill"""
```
Comment (almet) [inline budget/models.py]: In order to have a better readability, I would create a `prettify` function (inside this method) which will do the conversion if needed.

File: budget/models.py [hunk 2]
```diff
@@ -114,6 +130,19 @@ def get_bills(self):
...
             .order_by(Bill.id.desc())
 
+    def get_pretty_bills(self, export_format="json"):
+        """Return a list of project's bills with pretty formatting"""
+        return [{"what": bill.what,
+                 "amount": round(bill.amount, 2),
+                 "date": str(bill.date),
+                 "payer_name": Person.query.get(bill.payer_id).name,
+                 "payer_weight": Person.query.get(bill.payer_id).weight,
+                 "owers": [ower.name for ower in bill.owers]
```
Comment (almet) [inline budget/models.py]: Having an intermediary variable here for the owers would improve readability.

File: budget/templates/forms.html [hunk 3]
```diff
@@ -150,6 +150,18 @@
...
 
+{% macro export_project(form) %}
+    <fieldset>
+        <legend>{{ _("Export this project") }}</legend>
+    {{ form.hidden_tag() }}
+    {{ input(form.export_type) }}
+    {{ input(form.export_format) }}
+    </fieldset>
+    <div class="actions">
+        <button class="btn">{{ _("Export this project") }}</button>
```
Comment (almet) [inline budget/templates/forms.html]: For the action, consider: "Download this project's data"

File: budget/templates/layout.html [hunk 4]
```diff
@@ -51,6 +51,7 @@ <h1 class="col-2"><a class="navbar-brand" href="{{ url_for(".home") }}">#! money
                     <a href="#" class="nav-link dropdown-toggle" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><strong>{{ g.project.name }}</strong> {{ _("options") }} <b class="caret"></b></a>
                     <ul class="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
                         <li><a class="dropdown-item" href="{{ url_for(".edit_project") }}">{{ _("Project settings") }}</a></li>
+                        <li><a class="dropdown-item" href="{{ url_for(".export_project") }}">{{ _("Export project") }}</a></li>
```
Comment (almet) [inline budget/templates/layout.html]: I'm not sure where this should be: what about having a section in the project's settings instead?

File: budget/tests.py [hunk 5]
```diff
@@ -624,6 +624,116 @@ def test_settle_zero(self):
             self.assertNotEqual(0.0, rounded_amount,
                                 msg='%f is equal to zero after rounding' % t['amount'])
 
+    def test_export(self):
```
Comment (almet) [inline budget/tests.py]: Thank you so much for taking the time to write tests for this!

**Comment by 0livd** [comment]
Ok I amended the commit to improve readability and made some rewording. I also moved the export form to project's settings, the choice of a separate page was purely arbitrary :)

**Comment by almet** [comment]
This looks good to me!

**Pull request merged**